packages/server/src/context.ts

```diff
@@ -134,6 +134,27 @@ export function getTraceId(req: Request): string | undefined {
   return undefined;
 }
 
+export function buildTracingExtension(): Extension | undefined {
+  const ctx = getRequestContext();
```

Comment:
Yeah, that was one thing I wasn't sure about. Using `getRequestContext()` is what exposed a lot of tests that should have been using a `withTestContext`. I agree that using the `tryGetRequestContext` would definitely be the safer option for now even though it seems like this function could always expect to be run within a request context. Are there situations where this wouldn't be running in a request context that I'm not considering?

Reply:
Just chiming in here, but in my ideal world we would be able to assume that basically any code in `@medplum/server` has access to a request context

Reply:
> in my ideal world we would be able to assume that basically any code in @medplum/server has access to a request context

🤔  Hm, I see where you're coming from, but I don't believe that is inherently true.

1. There are cases that do not cleanly map to request contexts (i.e., app startup, crons).  We could try to establish more conventions (i.e., crons must initiate a pseudo request context), but
2. I'm concerned about context complexity.  It's increased burden on tests, increased burden on the engineer to juggle 

For example, last week when we had a bunch of test failures because the logging code required a request context. There was another case where we had to wrap `initApp` with a request context. IMHO, that's just incorrect.

You could say, "just use `withTestContext` everywhere", but that's also not perfect.  There are a bunch of mocked stubs in `withTestContext` (i.e., `{} as unknown as Login`).  In the spirit of "don't lie to the computer", I think the more honest model would be to acknowledge that there are cases that are simply not in a request.

This is probably worth getting alignment on.  Might need a longer offline discussion.

Reply:
FWIW, I went ahead and merged this using `tryGetRequestContext`. Regardless of the outcome of a larger discussion, this would not be the right place to enforce the existence of a request context.